src/uct/cuda/base/cuda_iface.c

```diff
@@ -173,18 +173,48 @@ ucs_status_t uct_cuda_base_iface_event_fd_arm(uct_iface_h tl_iface,
     return UCS_OK;
 }
 
+static UCS_F_ALWAYS_INLINE unsigned
+uct_cuda_base_progress_event_queue(uct_cuda_iface_t *iface,
+                                   ucs_queue_head_t *queue_head,
+                                   unsigned max_events)
+{
+    unsigned count = 0;
+    uct_cuda_event_desc_t *cuda_event;
+
+    ucs_queue_for_each_extract(cuda_event, queue_head, queue,
+                               cuEventQuery(cuda_event->event) ==
+                                       CUDA_SUCCESS) {
+        if (cuda_event->comp != NULL) {
+            ucs_trace_data("cuda_copy event %p completed", cuda_event);
+            uct_invoke_completion(cuda_event->comp, UCS_OK);
+        }
+
+        iface->ops->progress_event(&iface->super.super, cuda_event);
+
+        ucs_trace_poll("CUDA Event Done :%p", cuda_event);
+        ucs_mpool_put(cuda_event);
+        count++;
+        if (count >= max_events) {
```

Comment:
can add it to the condition in line 185:
`(count < max_events) && (cuEventQuery(cuda_event->event) == CUDA_SUCCESS)`